File: backend/src/main/java/com/people_manager/dto/request/UserRequestDTO.java
package com.people_manager.dto.request;

import com.people_manager.base.annotation.MandatoryField;
import com.people_manager.annotation.EmailValidate;
import com.people_manager.annotation.PasswordValidate;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public class UserRequestDTO {

    private Long id;
    private String name;

    private String login;

    @PasswordValidate(name = "Senha")
    private String password;

    private String confirmPassword;

    @EmailValidate(name = "Email")
    private String email;

    private Boolean enabled;
}

File: frontend/src/models/user/UserRequestDto.js
/**
 * DTO para requisição de criação/edição de usuário
 */
export class UserRequestDto {
    constructor(id, name, login, password, confirmPassword, email, enabled) {
        this.id = id; // opcional - usado apenas na edição
        this.name = name;
        this.login = login;
        this.password = password; // opcional na edição
        this.confirmPassword = confirmPassword; // opcional na edição
        this.email = email;
        this.enabled = enabled; // Boolean
    }
}
